ccm_web/server/src/invitation/cirrus-invitation.service.ts

```diff
@@ -0,0 +1,74 @@
+import axios from 'axios'
+import { randomUUID } from 'crypto'
+import FormData from 'form-data'
+import { lastValueFrom } from 'rxjs'
+import { HttpService } from '@nestjs/axios'
+import { ConfigService } from '@nestjs/config'
+import { Injectable } from '@nestjs/common'
+
+import { CirrusErrorData, CirrusInvitationResponse } from './cirrus-invitation.interfaces'
+import { InvitationAPIError } from './invitation.errors'
+
+import { Config } from '../config'
+import baseLogger from '../logger'
+
+const logger = baseLogger.child({ filePath: __filename })
+
+@Injectable()
+export class CirrusInvitationService {
+  url: string
+  entityID: string
+  sponsorName: string
+  key: string
+  secret: string
+
+  constructor (
+    private readonly configService: ConfigService<Config, true>,
+    private readonly httpService: HttpService
+  ) {
+    const invitationConfig = configService.get('invitation', { infer: true })
+    this.url = invitationConfig.apiURL
+    this.entityID = invitationConfig.apiEntityID
+    this.sponsorName = invitationConfig.apiSponsorName
+    this.key = invitationConfig.apiKey
+    this.secret = invitationConfig.apiSecret
+  }
+
+  async sendInvitations (userEmails: string[]): Promise<CirrusInvitationResponse | CirrusErrorData> {
+    if (userEmails.length === 0) {
+      throw new InvitationAPIError('Argument "userEmails" array is empty.')
+    }
```

Comment:
`InvitationAPIError` probably isn't the best description for this problem; that would be something related to calling the Cirrus API. Couple options here: You could create a new error subclass that does describe the problem (e.g. `InvitationError` or `InvitationInputError`), or you could just throw a plain `Error` with a good message passed to it (which would lead to an Internal Server Error, like any of these would.

Reply:
OK.  I'll go with the plain `Error` option.

Reply:
🏁